app/services/record_declaration.rb

```diff
@@ -205,6 +206,14 @@ def original_participant_declaration
     @original_participant_declaration ||= participant_declaration.duplicate_declarations.order(created_at: :asc).first
   end
 
+  def validate_not_retained_if_mentor
+    return unless declaration_type&.match?(/retained-\d/)
+    return unless participant_profile && participant_profile.mentor?
+    return unless cohort && cohort.mentor_funding?
```

Comment:
should this come first just to shorten the if statements we need to go through?

Reply:
Ah actually we can't do this as cohort delegates to participant profile -> schedule and if the profile is `nil` it throws an error